### PR TITLE
Feat/processing modal

### PR DESCRIPTION
**Pull request by Zueis**
[Feature]:
- Add the ConfirmStartProcessing component with the unit-test file.
- Display a warning message when the status of our bot for the analysis server is "inProgress".

[Changes]:
- Change 'any' types in code with related types.

[Fix]:
- Fix the issue in jest.config file and update package.JSON

### CODE REVIEW

**Comment by cyri113** [comment]
branches should be based on main (master). Right now you are committing all the code from the sentry issue too.

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,5 @@ yarn-error.log*
 next-env.d.ts
 
 .env
+
+.sentryclirc
```
Comment (cyri113) [inline .gitignore:40]: shouldn't be here

File: src/utils/types.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export type IToken = {
 export type IGuild = {
   readonly guildId: string;
   readonly guildName: string;
+  readonly _id?: string;
```
Comment (cyri113) [inline src/utils/types.ts:11]: what is this for?

File: src/components/global/FilterByChannels.tsx [hunk 3]
```diff
@@ -32,7 +30,7 @@ const FilterByChannels = ({
   const [guild, setGuild] = useState<IGuild>();
   const [channels, setChannels] = useState<Array<IGuildChannels>>([]);
   const [selectedChannels, setSelectedChannels] = useState<
-    { channelId: string; channelName: string }[]
+    Array<IChannelWithoutId>
```
Comment (cyri113) [inline src/components/global/FilterByChannels.tsx:33]: Nice! 👍🏽 

Does this ever include an `IChannel`?

File: src/components/pages/settings/ChannelSelection.tsx [hunk 4]
```diff
@@ -9,23 +9,36 @@ import { IoClose } from 'react-icons/io5';
...
   const [open, setOpen] = useState(false);
+  const [openProcessing, SetOpenProcessing] = useState(false);
+
   const [fullWidth, setFullWidth] = React.useState(true);
   const [guild, setGuild] = useState<IGuild>();
-  const [channels, setChannels] = useState<Array<any>>([]);
-  const [selectedChannels, setSelectedChannels] = useState<Array<any>>([]);
+  const [channels, setChannels] = useState<Array<IGuildChannels>>([]);
+  const [selectedChannels, setSelectedChannels] = useState<
+    Array<IChannelWithoutId>
```
Comment (cyri113) [inline src/components/pages/settings/ChannelSelection.tsx:40]: Will it ever include with a ChannelId?

File: src/components/pages/settings/ChannelSelection.tsx [hunk 5]
```diff
@@ -76,21 +97,20 @@ export default function ChannelSelection({ emitable, submit }: IProps) {
       }
     }
 
-    const result = [].concat(
-      ...channels.map((channel: any) => {
+    const result = ([] as { channelId: string; channelName: string }[]).concat(
```
Comment (cyri113) [inline src/components/pages/settings/ChannelSelection.tsx:100]: Can we Type this?

File: src/components/pages/settings/ChannelSelection.tsx [hunk 6]
```diff
@@ -103,36 +123,35 @@ export default function ChannelSelection({ emitable, submit }: IProps) {
...
-    setChannels((preChannels) => {
-      return preChannels.map((preChannel) => {
-        if (selectedGuild === preChannel.id) {
-          Object.keys(preChannel.selected).forEach((key: string) => {
-            preChannel.selected[key] = status;
-          });
-        }
-        return preChannel;
-      });
+    const updatedChannels = channels.map((channel) => {
```
Comment (cyri113) [inline src/components/pages/settings/ChannelSelection.tsx:137]: Can we Type the channel?

File: src/components/pages/settings/ChannelSelection.tsx [hunk 7]
```diff
@@ -144,38 +163,49 @@ export default function ChannelSelection({ emitable, submit }: IProps) {
       }
     }
 
-    const result = [].concat(
-      ...channels.map((channel: any) => {
+    const result = ([] as { channelId: string; channelName: string }[]).concat(
```
Comment (cyri113) [inline src/components/pages/settings/ChannelSelection.tsx:166]: can we Type this?

File: src/components/pages/settings/ConfirmStartProcessing.spec.tsx
Comment (cyri113) [inline src/components/pages/settings/ConfirmStartProcessing.spec.tsx:1]: Looks great for a first test!

**Review by cyri113 — CHANGES_REQUESTED** [review]

File: src/utils/types.ts (hunk 2 above)
Comment (Zueis) [reply src/utils/types.ts:11]: Sorry, I forgot to remove this.

File: .gitignore (hunk 1 above)
Comment (Zueis) [reply .gitignore:40]: Sorry, it was in my local so I put it in .gitignore manually. will remove it.

File: src/components/pages/settings/ChannelSelection.tsx (hunk 7 above)
Comment (Zueis) [reply src/components/pages/settings/ChannelSelection.tsx:168]: Actually its IChannel type without _id. So, I replace it everywhere.

File: src/components/pages/settings/ChannelSelection.tsx (hunk 6 above)
Comment (Zueis) [reply src/components/pages/settings/ChannelSelection.tsx:139]: done

File: src/components/global/FilterByChannels.tsx (hunk 3 above)
Comment (Zueis) [reply src/components/global/FilterByChannels.tsx:33]: Yes.

**Review by cyri113 — APPROVED** [review]

**Pull request merged**